dds/DCPS/transport/multicast/MulticastDataLink.cpp

```diff
@@ -296,10 +297,10 @@ MulticastDataLink::sample_received(ReceivedDataSample& sample)
 
       for (MulticastSessionMap::iterator it(temp_sessions.begin());
           it != temp_sessions.end(); ++it) {
-        it->second->control_received(sample.header_.submessage_id_,
-            sample.sample_);
+        bool ctrl_rcvd = it->second->control_received(sample.header_.submessage_id_,
+                                                      sample.sample_);
         // reset read pointer
-        if (ptr) {
+        if (ctrl_rcvd && ptr) {
```

Comment:
What's the rationale behind not resetting rd_ptr when control_received returns false?  Maybe this deserves clarification in the comment on ln 302 (or I'm missing something obvious).